out_writeapi_test.go

```diff
@@ -38,3 +46,221 @@ func TestFLBPluginRegister(t *testing.T) {
+	return m.GetWriteStreamFunc(ctx, req, opts...)
+}
+
+func (m *MockManagedWriterClient) Close() error {
+	return m.client.Close()
+}
+
+func (m *MockManagedWriterClient) BatchCommitWriteStreams(ctx context.Context, req *storagepb.BatchCommitWriteStreamsRequest, opts ...gax.CallOption) (*storagepb.BatchCommitWriteStreamsResponse, error) {
+	return m.client.BatchCommitWriteStreams(ctx, req, opts...)
+}
+
+func (m *MockManagedWriterClient) CreateWriteStream(ctx context.Context, req *storagepb.CreateWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+	return m.client.CreateWriteStream(ctx, req, opts...)
+}
+
+type StreamChecks struct {
+	calledGetContext int
+	createDecoder    int
+	gotRecord        int
+	appendRows       int
+	appendQueue      int
+	checkReady       int
+}
+
+type MockManagedStream struct {
+	managedstream  *managedwriter.ManagedStream
+	AppendRowsFunc func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error)
+	CloseFunc      func() error
+	FinalizeFunc   func(ctx context.Context, opts ...gax.CallOption) (int64, error)
+	FlushRowsFunc  func(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error)
+	StreamNameFunc func() string
+}
+
+func (m *MockManagedStream) AppendRows(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+	return m.AppendRowsFunc(ctx, data, opts...)
+}
+
+func (m *MockManagedStream) Finalize(ctx context.Context, opts ...gax.CallOption) (int64, error) {
+	return m.FinalizeFunc(ctx, opts...)
+}
+
+func (m *MockManagedStream) FlushRows(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error) {
+	return m.FlushRowsFunc(ctx, offset, opts...)
+}
+
+func (m *MockManagedStream) StreamName() string {
+	return m.StreamNameFunc()
+}
+
+func (m *MockManagedStream) Close() error {
+	return m.managedstream.Close()
+}
+
+func TestFLBPluginFlushCtx(t *testing.T) {
+	checks := new(StreamChecks)
+	var setID int
+	mockClient := &MockManagedWriterClient{
+		NewManagedStreamFunc: func(ctx context.Context, opts ...managedwriter.WriterOption) (*managedwriter.ManagedStream, error) {
+			return nil, nil
+
+		},
+		GetWriteStreamFunc: func(ctx context.Context, req *storagepb.GetWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+			return &storagepb.WriteStream{
+				Name: "mockstream",
+				TableSchema: &storagepb.TableSchema{
+					Fields: []*storagepb.TableFieldSchema{
+						{Name: "Time", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+						{Name: "Text", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+					},
+				},
+			}, nil
+		},
+	}
+
+	originalFunc := getClient
+	getClient = func(ctx context.Context, projectID string) (ManagedWriterClient, error) {
+		return mockClient, nil
+	}
+	defer func() { getClient = originalFunc }()
+
+	res := []bool{}
```

Comment:
Not clear what this array is for; please add a comment and/or a more descriptive name.

Reply:
This is connected to the first change where I append to simulate the results queue!